### PR TITLE
API Reference Documentation gen redux

### PR DESCRIPTION
**Pull request by moxiegirl**
Trying this merge again

### CODE REVIEW

File: src/auth/appConfig.ts
Comment (zone117x) [inline src/auth/appConfig.ts]: All of these `@type {...}` are jsdoc/flow leftovers, we can remove them so they don't show up as generic/unknown tags in the typedoc gen

File: src/auth/appConfig.ts
Comment (zone117x) [inline src/auth/appConfig.ts]: I originally suggested we use these `public`/`private` modifiers for a couple reasons:
1) They are a recommended eslint/tslint rule (currently not enabled), which we may want in the future.
2) I originally thought it was one of the only ways to control some parts of the docgen output (before finding out about the `@ignore` feature).

But now, I think we should hold off on introducing them for a couple reasons. Typescript enforces `private` so it controls how other typescript projects can use this lib, and `public` gives devs strong indication of intended usage/purpose. 

I think we'll be in a better position to do something like enable the eslint flag and go through the codebase to make these decisions once we have some of our other typescript repos (and 3rd parties) using this lib.

File: src/auth/authApp.ts
Comment (zone117x) [inline src/auth/authApp.ts]: These `@deprecated` tags and reference links are really handy! I think we should use these, but also keep the desc and param descriptions intact until we have the rest of tutorials/docs, app generators, and e2e tests changed to using the `UserSession` class. 

And maybe keep them until/unless we remove the functions all-together, since the available params and behavior between these functions and the `UserSession` versions may not be the same, or diverge, either intentionally or due to bugs.

File: src/profiles/profileSchemas/personLegacy.ts
Comment (zone117x) [inline src/profiles/profileSchemas/personLegacy.ts:3]: There are a few areas with empty `@param`s are these accidental?

**Review by zone117x — COMMENTED** [review]
Pulled down locally, cleaned and rebuilt, and seeing some items that don't seem like we want to expose? 

image

File: src/wallet.ts
Comment (zone117x) [inline src/wallet.ts]: Looks like the whitespacing here is causing the tags to be interpreted as a comments. Should be fixable by just removing the extra column of spaces.

Example: 
image

File: src/auth/appConfig.ts
Comment (moxiegirl) [reply src/auth/appConfig.ts]: OK. I'll make those changes.

File: src/auth/authApp.ts
Comment (moxiegirl) [reply src/auth/authApp.ts]: @zone117x I think that we should just remove them and leave the deprecated. The generated files have links to the code if they are using the existing call and want to maintain it -- a single click.  And the link to the `UserSession` is there if they got here incorrectly or want to go to the latest.

File: src/profiles/profileSchemas/personLegacy.ts
Comment (moxiegirl) [reply src/profiles/profileSchemas/personLegacy.ts:3]: These are on ignored calls that aren't in the docs.  So, I didn't spend time on filling them in.

File: src/auth/userSession.ts
Comment (zone117x) [inline src/auth/userSession.ts:39]: I noticed that links to type from external files are sometimes not generated. For example `GetFileOptions` here:

----

image

----

I found that using typescript's [`import() types syntax`](https://davidea.st/articles/typescript-2-9-import-types) fixes the issue. Example with `getFile(path: string, options?: import('../storage').GetFileOptions)`:

----

image

File: src/wallet.ts
Comment (moxiegirl) [reply src/wallet.ts]: Fixed.

File: src/auth/userSession.ts
Comment (moxiegirl) [reply src/auth/userSession.ts:39]: I'd like to fix this after the merge. I have a number of other things to do. This is just the first pass.

**Comment by moxiegirl** [comment]
Yeah, this wasn't happening before I rebased. Like I said, I need to do another pass for other things I want to do before we go live. I'll address this after the merge.

> 
> image

File: src/auth/authApp.ts
Comment (zone117x) [reply src/auth/authApp.ts]: Two main issues I'm concerned about here:

1. `UserSession` versions of these functions are not covered in our integration tests, and could have unknown or undocumented behavioral differences from the original. This has already happened with several of these functions. Leaving docs blank and linking to the other version implies that this is will not be the case. 

2. The `UserSession` versions can (and already have) intentionally diverged in usage & behavior from the original ones. For example, the current docs for `UserSession.redirectToSignIn(...)`, copied over from the original, are not really accurate. It tells devs to use other class methods (like `makeAuthRequest`) if they `require more fine grained control over how the authentication request`. Once updated to explain that class constructor options are the intended way to customize auth flow options, these empty docs with links to UserSession are _less_ than helpful, and will force devs to read through src to figure out what's going on. 

Removing these functions from the docs entirely would cause devs less problems and confusion, but we can't yet do that because all our other material still points devs to these functions. 

Seems reasonable to leave the existing, accurate docs for these intact, then we can eventually start removing things once they are effectively deprecated, rather than just tagged as such.

Comment (moxiegirl) [reply src/auth/authApp.ts]: 1. Isn't this a problem with how `UserSession` and the tests were implemented? If use of `UserSession` is problematic **we should fix the problems for `UserSessions`** . We deprecated the old functions, we say they are deprecated, we should not be documenting them. But if we haven't safely implemented the new functions why did we deprecate the old?

2. The fact that the UserSessions diverged seem fine to me --- they are new and the old ones are deprecated. Again, the problems with the `UserSession` docs should be *fixed if they are the recommended path forward* not left in place with problems we know exist.

**Comment by yknl** [comment]
To address the questions from the previous PR:

> Usersession.getOrSetLocalGaiaHubConnection not previously exposed should it be?

`getOrSetLocalGaiaHubConnection` should be private, I don't see a reason for developers to need to use this directly.

> Did we intend to deprecate the getFileURl cin storage/index.ts ?

The use of `getFileURl` directly is being deprecated but it should be used via UserSession in v19+

> As far as the include files go, right now I have most of them attached. I'm mulling whether they shouldn't simply be in the using docs and not in here at all.

Are you proposing that we remove the `includes/auth.md` and `includes/storage.md`?

> storage/index.ts has two functions encryptContent and decryptContent they are not putting out deprecation warnings, should they?

They were outputting deprecation warnings in v19. It doesn't look like the way it's exported has changed. So I would say it should be.

> Existing Javascript is not exposing the Profile1 object but we have examples using the toToken` method in there. I think we want this public

Is this referring to file:///Users/Yukan/Desktop/work/blockstack/blockstack.js/docs/classes/profile.html?

**Comment by yknl** [comment]
Do we want to show the Interfaces, Type alias, Variables and Object literals sections? They seem a bit irrelevant for most developers and might be confusing. I would also recommend leaving the social media service classes out of the index.

File: src/auth/authApp.ts
Comment (moxiegirl) [reply src/auth/authApp.ts]: OK. I added back in the content for the deprecated methods.

File: src/auth/userSession.ts
Comment (moxiegirl) [reply src/auth/userSession.ts:39]: This worked for the interface `GetFileOptions` but does not work for the type `PutFileOptions`.

**Comment by moxiegirl** [comment]
> getOrSetLocalGaiaHubConnection

Hidden.

@yknl thanks for your replies. Implemented as you suggested.

> Do we want to show the Interfaces, Type alias, Variables and Object literals sections? They seem a bit irrelevant for most developers and might be confusing. I would also recommend leaving the social media service classes out of the index

We are showing these in the other SDKS so we are including them as agreed on in Slack.

File: src/auth/appConfig.ts
Comment (zone117x) [inline src/auth/appConfig.ts]: Typo here an a couple lines below

File: src/auth/userSession.ts
Comment (zone117x) [inline src/auth/userSession.ts]: Looks like an accidental auto-import typo here

File: src/auth/userSession.ts
Comment (zone117x) [inline src/auth/userSession.ts]: Same here (looks like an accidental auto-import typo)

File: src/wallet.ts
Comment (zone117x) [inline src/wallet.ts:71]: Some `private` modifiers here an in a couple other files

File: package.json
Comment (zone117x) [inline package.json]: Looks like an accidental auto-import typo here

File: src/auth/userSession.ts
Comment (moxiegirl) [reply src/auth/userSession.ts]: This was a fix recommended by the linter to fix a problem.

File: src/auth/userSession.ts
Comment (moxiegirl) [reply src/auth/userSession.ts]: Again...eslint.  Do you want me to turn off the linter?

File: package.json
Comment (moxiegirl) [reply package.json]: See above comments on the linter.

File: src/auth/authApp.ts
Comment (zone117x) [inline src/auth/authApp.ts]: Hank recently fixed this bug and removed this field.

**Review by yknl — APPROVED** [review]

**Review by zone117x — APPROVED** [review]

**Pull request merged**
